api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto

```diff
@@ -108,6 +108,14 @@ message TcpProxy {
     repeated ClusterWeight clusters = 1 [(validate.rules).repeated = {min_items: 1}];
   }
 
+  // Configuration for tunneling TCP over other transports or application layers.
+  // Currently, only HTTP/2 is supported. When other options exist, HTTP/2 will
+  // remain the default.
```

Comment:
Can we future proof this by having a oneof that specifies a required protocol type, with the only type currently supported being something like `Http2TunnelConfig` or something like that? Then `hostname` can be embedded in that message. This will future proof adding additional protocols in the future if needed.

Reply:
HTTP/1.1 and HTTP/3 would both benefit from a configured as well, so I think it's orthogonal of transports for the forseeable future and the current structure makes sense.  Happy to move if you have thoughts of other things we want to tunnel over

Reply:
It's up to you, but if you think there will be non-HTTP transports in the future, you might still consider the structure I suggested, and just call the current option `HttpTunnelConfig`.

Reply:
I don't know of any, so I'd prefer to avoid the layer of indirection and risk the deprecation dance if you're OK either way.  If you have a preference it's not like it'll take more than 2 minutes though, so just let me know :-)

Reply:
Well, what I'm suggesting will avoid deprecation in the future, at the expense of a layer of indirection. I don't feel strongly about it so up to you.